src/Compilers/VisualBasic/Portable/Lowering/LocalRewriter/LocalRewriter_Query.vb

```diff
@@ -162,16 +149,57 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                 If propertyDefName.StartsWith("$"c, StringComparison.Ordinal) AndAlso Not String.Equals(propertyDefName, StringConstants.Group, StringComparison.Ordinal) Then
                     ' Nested compound variable.
                     Debug.Assert(propertyDefName.Equals(StringConstants.It) OrElse propertyDefName.Equals(StringConstants.It1) OrElse propertyDefName.Equals(StringConstants.It2))
-                    PopulateRangeVariableMapForAnonymousType(syntax, getCallOrPropertyAccess, rangeVariables, firstUnmappedRangeVariable)
-
+                    PopulateRangeVariableMapForAnonymousType(syntax, getCallOrPropertyAccess, rangeVariables, firstUnmappedRangeVariable, rangeVariableMap, inExpressionLambda)
                 Else
                     Debug.Assert(IdentifierComparison.Equals(propertyDefName, rangeVariables(firstUnmappedRangeVariable).Name))
-                    _rangeVariableMap.Add(rangeVariables(firstUnmappedRangeVariable), getCallOrPropertyAccess)
+                    rangeVariableMap.Add(rangeVariables(firstUnmappedRangeVariable), getCallOrPropertyAccess)
                     firstUnmappedRangeVariable += 1
                 End If
             Next
         End Sub
 
+        Friend Shared Function RewriteQueryLambdaBody(
+            expressionRewriter As Func(Of BoundExpression, BoundExpression),
+            originalNode As BoundQueryLambda,
+            rangeVariableMap As Dictionary(Of RangeVariableSymbol, BoundExpression),
+            Optional instrumenterOpt As Instrumenter = Nothing,
+            Optional instrumentQueryLambdaBody As Boolean = False) As BoundStatement
+
+            Dim returnstmt As BoundStatement = New BoundReturnStatement(originalNode.Syntax,
+                                                                        expressionRewriter(originalNode.Expression),
+                                                                        Nothing,
+                                                                        Nothing)
+
+            If instrumentQueryLambdaBody Then
+                returnstmt = instrumenterOpt.InstrumentQueryLambdaBody(originalNode, returnstmt)
+            End If
+
+            For Each rangeVar As RangeVariableSymbol In originalNode.RangeVariables
```

Comment:
>For Each rangeVar As RangeVariableSymbol In originalNode.RangeVariables [](start = 12, length = 71)

This should be done in where we rewrite Expression, given that I suggested to move that part into the caller, this part should be moved there too. Consider extracting into its own method if you want to reuse the code.